docs/01-app/03-api-reference/04-functions/generate-metadata.mdx

```diff
@@ -1202,6 +1204,69 @@ Streaming metadata improves perceived performance by reducing [TTFB](https://dev
 
 Overriding `htmlLimitedBots` could lead to longer response times. Streaming metadata is an advanced feature, and the default should be sufficient for most cases.
 
+### With Cache Components
+
+When [Cache Components](/docs/app/getting-started/cache-components) is enabled, `generateMetadata` follows the same rules as other components. If metadata accesses runtime data (`cookies()`, `headers()`, `params`, `searchParams`) or performs uncached data fetching, it defers to request time.
+
+How Next.js handles this depends on the rest of your page:
+
+- **If other parts also defer to request time**: Prerendering generates a static shell, and metadata streams in with other deferred content.
+- **If the page or layout is otherwise fully prerenderable**: Next.js requires an explicit choice: cache the data if possible, or signal that deferred rendering is intentional.
+
+Streaming metadata at runtime while the rest of the page is fully prerenderable is not common. To ensure this behavior is intentional, an error is raised indicating which page or layout needs to be handled.
+
+To resolve this, you have two options. If metadata depends on external data but not runtime data, use `use cache`:
+
+```tsx filename="app/blog/[slug]/page.tsx" highlight={2}
+export async function generateMetadata({ params }) {
+  'use cache'
+  const { slug } = await params
+  const post = await db.getPost(slug)
+  return { title: post.title }
+}
+```
+
+If metadata genuinely requires runtime data, add a dynamic marker component to your page:
+
+```tsx filename="app/page.tsx" highlight={11-14,19,28}
+import { Suspense } from 'react'
+import { cookies } from 'next/headers'
+import { connection } from 'next/server'
+
+export async function generateMetadata() {
+  const token = (await cookies()).get('token')?.value
+  // ... use token to fetch personalized metadata
+  return { title: 'Personalized Title' }
+}
+
+const Connection = async () => {
+  await connection()
```

Comment:
I guess you can't add `connection` in `generateMetadata`?

Reply:
Not to get around this guidance